client/state/sharing/publicize/actions.js

```diff
@@ -59,14 +61,39 @@ export function sharePost( siteId, postId, skippedConnections, message ) {
  */
 export function fetchConnections( siteId ) {
 	return ( dispatch ) => {
-		dispatch( {
-			type: PUBLICIZE_CONNECTIONS_REQUEST,
-			siteId,
-		} );
+		dispatch( requestConnections( siteId ) );
 
 		return wpcom.undocumented().siteConnections( siteId )
-			.then( ( response ) => dispatch( receiveConnections( siteId, response ) ) )
-			.catch( ( error ) => dispatch( failConnectionsRequest( siteId, error ) ) );
+			.then( ( connections ) => {
+				dispatch( receiveConnections( siteId, connections ) );
+				dispatch( requestConnectionsSuccess( siteId ) );
+			} )
+			.catch( ( error ) => dispatch( requestConnectionsFailure( siteId, error ) ) );
+	};
+}
+
+/**
+ * Triggers a network request to request a Publicize connection for the
+ * specified site ID.
+ *
+ * @param  {Number} siteId       Site ID
+ * @param  {Number} connectionId ID of the connection to be fetched.
+ * @return {Function}            Action thunk
+ */
+export function fetchConnection( siteId, connectionId ) {
+	return ( dispatch ) => {
+		dispatch( requestConnections( siteId ) );
```

Comment:
That's correct, I originally suggested 2 action types in  https://github.com/Automattic/wp-calypso/pull/9273#discussion_r87506843

See also `requestSite` vs `requestSites` one dispatches `SITE_REQUEST` while the other dispatches `SITES_REQUEST`. Without the differentiation it can be difficult to know which endpoint we're requesting, which is useful in query components like `<QuerySites \>`